### ISSUE TITLE
Does not work with Javascript keyed collection Map

### ISSUE THREAD

**Issue by noahehall** [rtfeldman/seamless-immutable#171]
I am using a javacript map to store data I retrieve from an api for a udacity project.

However, the data is not being saved to the immutable object.

here is a branch you can clone and verify:
1. clone https://github.com/noahehall/udacity-trainschedule.git
2. npm install
3. npm run start

here is picture of implementation
<img width="985" alt="screen shot 2016-11-13 at 9 51 18 pm" src="https://cloud.githubusercontent.com/assets/10324554/20254674/8c14d2dc-a9eb-11e6-91d1-23ec4d26364c.png">

here is picture of error received from immutable
<img width="525" alt="screen shot 2016-11-13 at 9 53 16 pm" src="https://cloud.githubusercontent.com/assets/10324554/20254694/b12635c0-a9eb-11e6-917b-8687ce291c59.png">
 error TypeError: immutableDoesNotWorkWithJavascriptMaps.data.has is not a function(…)

Is there a workaround to get immutable to use Javascript maps, or should I abandon seamless immutable?

**Comment by noahehall** [comment]
hmm ive actually switched form using Maps with seamless-immutable, but will be interested to know if they will be supported in the future

**Comment by rtfeldman** [comment]
Happy to accept a PR for them!

**Comment by noahehall** [comment]
Definitely, ill take a swing at it. I switched to storing everything as an object for now

**Comment by gavin-gmlab** [comment]
I'm having this problem also. 

Seems like a .set still works, but a .get does not get the object back properly.

Tried a workaround to create a new map object => newMap = new Map(state.oldMap._mapData) seems to work. Not sure if there are other implications

**Comment by noahehall** [comment]
yeah sorry guys still havent had time to work on this, things at work have been crazy

**Comment by noahehall** [comment]
Finally got some time, picking this up ;)

**Comment by noahehall** [comment]
so i got drunk last night and worked on integrating maps,

you can see a preview here:
https://github.com/noahehall/seamless-immutable/tree/feature/map-and-weakmap

i'll need to do some clean up and refacter it a bit, add some tests, then ill submit a pull request

**Comment by mikeaustin** [comment]
Soo.. are Maps and Sets supported now? I'd digging around trying to find out.

**Comment by tedlin182** [comment]
@noahehall Any progress on this?
